File: P1A11_Termometro/C++/ArchivosO/Termometro.cpp
#include <iostream>
#include <iomanip>
#include <cstdlib>
#include <ctime>
#include "../ArchivosE/Termometro.h" 

using namespace std;

// Constructor por defecto: temperatura aleatoria entre 0 y 100
Termometro::Termometro() {
    // Inicializar la semilla aleatoria (en un programa real, esto debería hacerse una sola vez en main, 
    // pero para este ejemplo autocontenido lo hacemos aquí con cuidado o usamos un generador moderno)
    // Para simplificar y seguir el estilo académico:
    double aleatorio = (double)rand() / RAND_MAX; // 0.0 a 1.0
    temperatura = aleatorio * 100.0;
}

// Constructor parametrizado
Termometro::Termometro(double temp) {
    temperatura = temp;
}

double Termometro::TemperaturaC() {
    return temperatura;
}

double Termometro::TemperaturaF() {
    return (temperatura * 9.0 / 5.0) + 32.0;
}

double Termometro::TemperaturaK() {
    return temperatura + 273.15;
}

void Termometro::MostrarTemperaturas() {
    cout << fixed << setprecision(2);
    cout << "Temperatura en Celsius:    " << TemperaturaC() << " °C" << endl;
    cout << "Temperatura en Fahrenheit: " << TemperaturaF() << " °F" << endl;
    cout << "Temperatura en Kelvin:     " << TemperaturaK() << " K" << endl;
}

File: P1A11_Termometro/C#/Termometro/Termometro.cs
using System;

namespace Termometro
{
    public class Termometro
    {
        private double temperatura;

        // Constructor por defecto: temperatura aleatoria entre 0 y 100
        public Termometro()
        {
            Random random = new Random();
            temperatura = random.NextDouble() * 100;
        }

        // Constructor parametrizado: recibe temperatura en Celsius
        public Termometro(double temp)
        {
            temperatura = temp;
        }

        public double TemperaturaC()
        {
            return temperatura;
        }

        public double TemperaturaF()
        {
            return (temperatura * 9 / 5) + 32;
        }

        public double TemperaturaK()
        {
            return temperatura + 273.15;
        }

        public void MostrarTemperaturas()
        {
            Console.WriteLine($"Temperatura en Celsius:    {TemperaturaC():F2} °C");
            Console.WriteLine($"Temperatura en Fahrenheit: {TemperaturaF():F2} °F");
            Console.WriteLine($"Temperatura en Kelvin:     {TemperaturaK():F2} K");
        }
    }
}
